### ISSUE TITLE
Reorder account list when desired

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#1161]
We should offer a feature for users to rearrange the order of their list of accounts.

This would appear on the choose-account page (or after #1038, might be an option in the choose-account modal and takes you to a separate "manage accounts" page). The general shape of the UI should be:
* First the user indicates they want to manage the list of accounts.
* Then the items in the account list become draggable, for the user to move them around as they please.
* In that mode, there's some way to say "ok, done" and return to the normal version of the UI, where tapping on an item navigates to that account and they're not draggable.

### Implementation

The key ingredient for implementing this UI is [ReorderableListView](https://api.flutter.dev/flutter/material/ReorderableListView-class.html), from Flutter's Material implementation.

(In principle it'd be good to have a design for this, which might change some of the details I described above. But this is a feature that people use once, briefly; it makes their Zulip experience better in an ongoing way because now their accounts are in an order that makes sense for them; but then may not touch the feature itself again for a long time. So I think it's not a high priority to spend design effort on, and we should just implement something with the off-the-shelf ReorderableListView.)

**Comment by gnprice** [comment]
(In the legacy app, we instead constantly reorder the list automatically: each time you use an account, it moves to the top of the list. Today we [heard from beta users](https://chat.zulip.org/#narrow/channel/48-mobile/topic/switch-account.20UI/near/2004012) who are glad the new app doesn't do that.)

**Comment by khaeru** [comment]
Hi there, 

Thanks for the work on this responsive and clean app! 

I can't tell if the following is related to this issue, so leaving a comment here. Please let me know if it should be a separate issue. 

On the "Switch account" screen, I see two accounts:

1. Account A (test account on chat.zulip.com)
2. Account B (account on a *.zulipchat.com instance) 

These were created and added this order. 

Whenever I open the app, I first am shown new messages etc. for Account A. I must always switch to Account B before using it. I cannot find any way to open Account B automatically or by default.

What's not clear is whether the **display order** in the "Switch account" screen also determines the **account first shown on app (re)start**. In some other apps (like muli-account IMAP clients) these two things can be configured separately; but IMO it would be fine if in the Zulip app the latter is determined by the former, i.e. the account displayed first in the list would be the one that is opened on app start. If that is/will be the case, then I guess this issue would solve my problem, and I would suggest some visual cue/hint text about what effect(s) reordering the list will have.

**Comment by chrisbobbe** [comment]
I think #524 will help with your frustration at always landing on Account A when the app starts up. We've done the work for #524, we just haven't released it yet :) so look for it in the next release within probably a week or two.
